src/test/java/de/retest/web/selenium/ByBestMatchToRetestIdTest.java

```diff
@@ -0,0 +1,79 @@
+package de.retest.web.selenium;
+
+import static org.assertj.core.api.AssertionsForClassTypes.assertThat;
+import static org.assertj.core.api.AssertionsForClassTypes.assertThatThrownBy;
+import static org.mockito.Mockito.mock;
+
+import java.util.Arrays;
+
+import org.junit.jupiter.api.Test;
+
+import de.retest.ui.descriptors.Attributes;
+import de.retest.ui.descriptors.Element;
+import de.retest.ui.descriptors.IdentifyingAttributes;
+import de.retest.ui.descriptors.RootElement;
+
+class ByBestMatchToRetestIdTest {
+
+	@Test
+	void constructor_should_throw_new_illegal_argument_exception() {
+		assertThatThrownBy( () -> new ByBestMatchToRetestId( "" ) ) //
+				.isInstanceOf( IllegalArgumentException.class ) //
+				.hasMessage( "retestId must not be empty." );
+	}
+
+	@Test
+	void find_elements_should_throw_new_no_element_with_retest_id_found_exception() {
+		final RootElement rootElement = mock( RootElement.class );
+		assertThatThrownBy( () -> new ByBestMatchToRetestId( "foo" ).findElement( rootElement, rootElement ) )
+				.isInstanceOf( NoElementWithReTestIdFoundException.class ) //
+				.hasMessage( "No element with retest ID 'foo' found!" );// )
+	}
+
+	@Test
+	void find_elements_should_return_an_element_with_the_same_retest_id() {
+		final IdentifyingAttributes identifyingAtt = mock( IdentifyingAttributes.class );
+		final Attributes attributes = mock( Attributes.class );
+		final Element element = new Element( "foo", identifyingAtt, attributes );
+		final RootElement lastExpected =
+				new RootElement( "bar", identifyingAtt, attributes, null, Arrays.asList( element ), null, 0, null );
+		final RootElement lastActual =
+				new RootElement( "he", identifyingAtt, attributes, null, Arrays.asList( element ), null, 0, null );
+		final ByBestMatchToRetestId cut = new ByBestMatchToRetestId( "foo" );
+		assertThat( cut.findElement( lastExpected, lastActual ) ).isEqualTo( element );
+	}
+
+	@Test
+	void hash_code_should_return_an_expected_value() {
+		final ByBestMatchToRetestId cut = new ByBestMatchToRetestId( "foo" );
+		final int expected = 101574;
```

Comment:
You can do something like: 
`String foo = "foo";`
`final int expected = foo.hashCode();`